quesma/queryparser/pancake_model.go

```diff
@@ -0,0 +1,105 @@
+// Copyright Quesma, licensed under the Elastic License 2.0.
+// SPDX-License-Identifier: Elastic-2.0
+package queryparser
+
+import (
+	"quesma/logger"
+	"quesma/model"
+)
+
+type pancakeAggregationTopLevel struct {
+	children    []*pancakeAggregationLevel
+	whereClause model.Expr
+}
+
+type pancakeAggregationLevel struct {
+	name            string
+	queryType       model.QueryType
+	selectedColumns []model.Expr
+
+	// only for bucket aggregations
+	children []*pancakeAggregationLevel
```

Comment:
Maybe embedding `pancakeAggregationTopLevel` would be better and more clearly show intentions
```suggestion
	pancakeAggregationTopLevel
```